src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/LengthConverter.cs

```diff
@@ -143,23 +137,22 @@ public override bool CanConvertTo(ITypeDescriptorContext typeDescriptorContext,
         {
             ArgumentNullException.ThrowIfNull(destinationType);
 
-            if (    value != null
-                &&  value is double )
+            if (value is double doubleValue)
             {
-                double l = (double)value;
                 if (destinationType == typeof(string)) 
                 { 
-                    if(double.IsNaN(l)) 
-                        return "Auto";
+                    if(double.IsNaN(doubleValue)) 
+                        return NaNValue.ToString();
```

Comment:
You should keep the string literal here because I think your changes here will regress because it now allocates a string.

Reply:
Haha, you're totally right, I just copied and overlooked afterwards, this would be a major regression. Thank you!